### PR TITLE
fix: Don't throw error on create hypertable, if it exists already

### LINKED ISSUE #7695: bug in migrations, when restarting a node w/o unsafe-reset all
A possible bug has been found in migrations, when we restart node w/o unsafe-reset all:

```
Feb 24 18:53:49 n06.testnet.vega.xyz visor[2441191]: 2023-02-24T18:53:49.534Z        INFO        datanode.start.persistentPre.db-migrate        v3@v3.6.1/migration_sql.go:30        Executing statement: SELECT create_hypertable('rewards', 'vega_time', chunk_time_interval => INTERVAL '1 day', migrate_data => true);
Feb 24 18:53:49 n06.testnet.vega.xyz visor[2441191]: 
Feb 24 18:53:49 n06.testnet.vega.xyz visor[2441191]: 2023-02-24T18:53:49.535Z        INFO        datanode.start.persistentPre.db-migrate        v3@v3.6.1/migration_sql.go:32        Rollback transaction
Feb 24 18:53:49 n06.testnet.vega.xyz visor[2441191]: 2023-02-24T18:53:49.535Z        ERROR        datanode.start.persistentPre        start/node_pre.go:141        Failed to initialise database, retrying...        {"error": "failed to migrate to latest schema:error migrating sql schema: ERROR 0004_reward_history.sql: failed to run SQL migration: failed to execute SQL query \"SELECT create_hypertable('rewards', 'vega_time', chunk_time_interval => INTERVAL '1 day', migrate_data => true);\\n\": ERROR: table \"rewards\" is already a hypertable (SQLSTATE TS110)"}
```

Config looks ok:

```
GatewayEnabled = true
AutoInitialiseFromNetworkHistory = false
ChainID = "vega-fairground-202302221129"
MaxMemoryPercent = 33

...
...

[SQLStore]
  WipeOnStartup = false
  Level = "Info"
  UseEmbedded = false
  FanOutBufferSize = 1000
  RetentionPeriod = "forever"
  VerboseMigration = true
  [SQLStore.ConnectionConfig]
    Host = "localhost"
    Port = 5432
    Username = "vega"
    Password = "8tKhPqJEvMmndx5Gb5fLsTNeWD4w2obNAjkeFzPWKezpi5SkxYXut"
    Database = "data_node_db"
    SocketDir = "/tmp"
    MaxConnLifetime = "30m0s"
    MaxConnLifetimeJitter = "5m0s"
    MaxConnPoolSize = 0
    MinConnPoolSize = 0
```

Blocks and snapshots for DN&Vega looks also well:

DN:

```
2023-02-24T18:58:19.102Z	INFO	root	commands/last-block.go:82	Last block	{"height": 252357}
vega:
root@n06:/home/daniel# /home/vega/vegavisor_home/current/vega tools snapshot --db-path /home/vega/vega_home/state/node/snapshots/
Snapshots available: 10
	Height: 252000, Version: 252, Size 125, Hash: c35fd1f3c544c712a02dfe680102d5be1440ad75e5109e963e4a1d0414d2ad43
	Height: 251000, Version: 251, Size 125, Hash: d1565475a6a87ed6ac8219b2e02afdba6e8003465bfe227f94c71d1297b49b60
	Height: 250000, Version: 250, Size 125, Hash: 650dd677d97386f605b109731beba304052b0e733e2d23c320aac17f92dd4bea
```

### Additonal information:

- https://vegaprotocol.slack.com/archives/CEE07A084/p1677265345484559?thread_ts=1677261566.628979&cid=CEE07A084

### PR DESCRIPTION
**Pull request by zale144**
Closes #7695

### CODE REVIEW

File: datanode/sqlstore/migrations/0001_initial.sql [hunk 1]
```diff
@@ -1039,7 +1039,7 @@ CREATE TABLE positions(
   primary key (party_id, market_id, vega_time)
 );
 
-select create_hypertable('positions', 'vega_time', chunk_time_interval => INTERVAL '1 day');
+select create_hypertable('positions', 'vega_time', chunk_time_interval => INTERVAL '1 day', if_not_exists => true);
```
Comment (wwestgarth) [inline datanode/sqlstore/migrations/0001_initial.sql:1042]: If we're putting `if not exists` in here I guess we should also have them in `CREATE TABLE positions` too.

File: datanode/sqlstore/migrations/0004_reward_history.sql [hunk 2]
```diff
@@ -2,4 +2,4 @@
 
 ALTER TABLE rewards DROP CONSTRAINT IF EXISTS rewards_party_id_fkey;
 
-SELECT create_hypertable('rewards', 'vega_time', chunk_time_interval => INTERVAL '1 day', migrate_data => true);
+SELECT create_hypertable('rewards', 'vega_time', chunk_time_interval => INTERVAL '1 day', if_not_exists => true, migrate_data => true);
```
[the remark this answers is not in the document]
Comment (guoguojin) [reply datanode/sqlstore/migrations/0004_reward_history.sql:5]: You are right @wwestgarth and I think I'm to blame for the missing goose down, apologies for that.

@zale144 can we add the goose down in your PR?

**Review by guoguojin — CHANGES_REQUESTED** [review]

File: datanode/sqlstore/migrations/0004_reward_history.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (zale144) [reply datanode/sqlstore/migrations/0004_reward_history.sql:5]: @wwestgarth @guoguojin question, in `postgresql.conf` do we have this bit:
`shared_preload_libraries = 'timescaledb'` ?

[the remark this answers is not in the document]
Comment (fkondej) [reply datanode/sqlstore/migrations/0004_reward_history.sql:5]: Yes, we do have it on our machines.
Screenshot 2023-03-09 at 16 42 26

[the remark this answers is not in the document]
Comment (fkondej) [reply datanode/sqlstore/migrations/0004_reward_history.sql:5]: In our [docs](https://docs.vega.xyz/testnet/node-operators/migration-guides/upgrade-node), we ask node operators to run `timescaledb-tune`. That application makes sure this value is set properly. You can see an example [here](https://docs.timescale.com/timescaledb/latest/how-to-guides/configuration/timescaledb-tune/).

[the remark this answers is not in the document]
Comment (zale144) [reply datanode/sqlstore/migrations/0004_reward_history.sql:5]: Then if `show shared_preload_libraries;` outputs `timescaledb` , other than running migrations from a non-owner user, I have no other ideas 🤷

File: datanode/sqlstore/migrations/0001_initial.sql (hunk 1 above)
Comment (EVODelavega) [reply datanode/sqlstore/migrations/0001_initial.sql:1042]: The `goose Down` step has a `DROP TABLE positions cascade` statement, so if such a table were to exist, I'd assume something is wrong, and I'd expect to see an error in that case

Comment (wwestgarth) [reply datanode/sqlstore/migrations/0001_initial.sql:1042]: Yea I agree, I think the existence checks will just hide errors that we should just fix.

**Review by wwestgarth — APPROVED** [review]

**Review by guoguojin — APPROVED** [review]

**Pull request merged**